### PR TITLE
User to aggregatemetric notification

### PR DESCRIPTION
**Pull request by mgrauer**
@cpatrick PTAL. 

The backend to setting up aggregate metric notifications.  Some of it anyway.

### CODE REVIEW

File: modules/tracker/database/pgsql/1.2.4.sql [hunk 1]
```diff
@@ -117,7 +117,9 @@ CREATE INDEX "tracker_aggregate_metric_spec_producer_id" ON "tracker_aggregate_m
 CREATE INDEX "tracker_aggregate_metric_spec_branch" ON "tracker_aggregate_metric_spec" ("branch");
 
 CREATE TABLE IF NOT EXISTS "tracker_user2aggregate_metric_spec" (
+    "id" serial PRIMARY KEY,
```
Comment (cpatrick) [inline modules/tracker/database/pgsql/1.2.4.sql]: This is counter to our general convention of `<table_name_id>`.

File: modules/tracker/database/sqlite/1.2.4.sql [hunk 2]
```diff
@@ -115,7 +115,9 @@ CREATE INDEX IF NOT EXISTS "tracker_aggregate_metric_spec_producer_id" ON "track
 CREATE INDEX IF NOT EXISTS "tracker_aggregate_metric_spec_branch" ON "tracker_aggregate_metric_spec" ("branch");
 
 CREATE TABLE IF NOT EXISTS "tracker_user2aggregate_metric_spec" (
+    "id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
```
Comment (cpatrick) [inline modules/tracker/database/sqlite/1.2.4.sql]: This is counter to our general convention of `<table_name>_id`.

File: modules/tracker/database/pgsql/1.2.4.sql (hunk 1 above)
Comment (jamiesnape) [inline modules/tracker/database/pgsql/1.2.4.sql]: Does the MySQL table have a primary key? If not, this is correct.

**Comment by cpatrick** [comment]
Other than the nits (which may be illegitimate as @jamiesnape suggests), LGTM.

**Comment by mgrauer** [comment]
>Does the MySQL table have a primary key?

It has a primary compound key

    PRIMARY KEY (`user_id`, `aggregate_metric_spec_id`)


Looking at this more carefully, I don't have the corresponding unique compound indices in pgsl and sqlite.

Perhaps the way forward is to
* remove `tracker_user2aggregate_metric_spec`'s primary compound key, make it a unique compound key instead (so as to uphold what @jamiesnape says about `id` in the other two dbs for when no primary key exists in mysql)
* add a unique compound key to pgsql and sqlite

Thoughts?

**Comment by cpatrick** [comment]
I guess whatever we normally do in our `user2*` tables?

**Comment by mgrauer** [comment]
>normally

Ahem.

I'll follow `feed2community` and make a consistency fix for `user2group`.

**Comment by mgrauer** [comment]
@cpatrick PTAL.

This changeset is small since your last checkpoint.

**Comment by cpatrick** [comment]
LGTM.

**Pull request merged**